s3stream/src/main/java/com/automq/stream/s3/compact/StreamObjectCompactor.java

```diff
@@ -184,7 +184,7 @@ static boolean checkObjectGroupCouldBeCompact(List<S3ObjectMetadata> objectGroup
         if (objectGroup.size() == 1 && (MINOR.equals(compactionType) || MAJOR.equals(compactionType) || MINOR_V1.equals(compactionType))) {
             return false;
         }
-        if (CLEANUP_V1.equals(compactionType)) {
+        if (CLEANUP_V1.equals(compactionType) || MAJOR_V1.equals(compactionType)) {
```

Comment:
If objectsGroup has 100 objects, but the first composite object does not have dirty data, then MAJOR_V1 will not run.

Reply:
The expected behavior should be: do not run MAJOR_V1 compaction if objectGroup.size() == 1.

Reply:
got it. I'll change the code